plugins/inputs/jti_openconfig_telemetry/openconfig_telemetry.go

```diff
@@ -0,0 +1,324 @@
+	ClientId        string
+	Sensors         []string
+	SampleFrequency uint32
+	CertFile        string
+	Debug           bool
+	StrAsTags       bool
+
+	parser parsers.Parser
+	sync.Mutex
+
+	// keep the accumulator internally:
+	acc            telegraf.Accumulator
+	grpcClientConn *grpc.ClientConn
+}
+
+var sampleConfig = `
+  ## Device address to collect telemetry from
+  server = "localhost:1883"
+
+  ## Authentication details. Username and password are must if device expects 
+  ## authentication. Client ID must be unique when connecting from multiple instances 
+  ## of telegraf to the same device
+  username = "user"
+  password = "pass"
+  clientId = "telegraf"
+
+  ## Frequency to get data in milliseconds
+  sampleFrequency = 1000
+
+  ## Sensors to subscribe for
+  ## A identifier for each sensor can be provided in path by separating with space
+  ## Else sensor path will be used as identifier
+  ## When identifier is used, we can provide a list of space separated sensors. 
+  ## A single subscription will be created with all these sensors and data will 
+  ## be saved to measurement with this identifier name
+  sensors = [
+   "/interfaces/",
+   "collection /components/ /lldp",
+  ]
+
+  ## x509 Certificate to use with TLS connection. If it is not provided, an insecure 
+  ## channel will be opened with server
+  certFile = "/path/to/x509_cert_file"
+
+  ## To see data being received from gRPC server, set debug to true
+  debug = true
+
+  ## To treat all string values as tags, set this to true
+  strAsTags = false
+`
+
+func (m *OpenConfigTelemetry) SampleConfig() string {
+	return sampleConfig
+}
+
+func (m *OpenConfigTelemetry) Description() string {
+	return "Read JTI OpenConfig Telemetry from listed sensors"
+}
+
+func (m *OpenConfigTelemetry) SetParser(parser parsers.Parser) {
+	m.parser = parser
+}
+
+func (m *OpenConfigTelemetry) Gather(acc telegraf.Accumulator) error {
+	return nil
+}
+
+func (m *OpenConfigTelemetry) Stop() {
+	m.Lock()
+	defer m.Unlock()
+	m.grpcClientConn.Close()
+}
+
+// Takes in XML path with predicates and returns list of tags+values along with a final XML path without predicates
+func spitTagsNPath(xmlpath string) (string, map[string]string) {
+	re := regexp.MustCompile("\\/([^\\/]*)\\[([A-Za-z0-9\\-\\/]*)\\=([^\\[]*)\\]")
+	subs := re.FindAllStringSubmatch(xmlpath, -1)
+	tags := make(map[string]string)
+
+	// Given XML path, this will spit out final path without predicates
+	if len(subs) > 0 {
+		for _, sub := range subs {
+			tagKey := strings.Split(xmlpath, sub[0])[0]
+			tagKey += "/" + strings.TrimSpace(sub[1]) + "/@" + strings.TrimSpace(sub[2])
+			tagValue := strings.Replace(sub[3], "'", "", -1)
+
+			tags[tagKey] = tagValue
+			xmlpath = strings.Replace(xmlpath, sub[0], "/"+strings.TrimSpace(sub[1]), 1)
+		}
+	}
+
+	return xmlpath, tags
+}
+
+func (m *OpenConfigTelemetry) Start(acc telegraf.Accumulator) error {
+	log.Print("I! Started JTI OpenConfig Telemetry plugin\n")
+
+	m.Lock()
```

Comment:
You may be able to remove this lock, Telegraf guarantees it will serialize calls to Start(), Stop(), etc.

Reply:
Removed it.